File: 64-medium-signal-path/go/main.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

type input struct {
	N     int     `json:"n"`
	Root  int     `json:"root"`
	Pnl   []int64 `json:"pnl"`
	Left  []int   `json:"left"`
	Right []int   `json:"right"`
}

func main() {
	var in input
	if err := json.NewDecoder(os.Stdin).Decode(&in); err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(1)
	}
	fmt.Println(solve(in.N, in.Root, in.Pnl, in.Left, in.Right))
}

File: 64-medium-signal-path/rust/src/main.rs
use signal_path::solve;
use signal_path::Input;
use std::io::{self, Read};

fn main() {
    let mut buf = String::new();
    io::stdin().read_to_string(&mut buf).unwrap();
    let inp: Input = serde_json::from_str(&buf).unwrap();
    println!(
        "{}",
        solve(inp.n, inp.root, &inp.pnl, &inp.left, &inp.right)
    );
}
